p2p/src/main/java/bisq/network/p2p/storage/P2PDataStorage.java

```diff
@@ -390,50 +390,64 @@ public boolean addProtectedStorageEntry(ProtectedStorageEntry protectedStorageEn
             return false;
         }
 
-        boolean sequenceNrValid = isSequenceNrValid(protectedStorageEntry.getSequenceNumber(), hashOfPayload);
-        boolean result = sequenceNrValid &&
-                checkPublicKeys(protectedStorageEntry, true)
-                && checkSignature(protectedStorageEntry);
+        // TODO: Combine with hasSequenceNrIncreased check, but keep existing behavior for now
+        if(!isSequenceNrValid(protectedStorageEntry.getSequenceNumber(), hashOfPayload) ) {
+            log.trace("addProtectedStorageEntry failed due to invalid sequence number");
+
+            return false;
+        }
+
+        // Verify the ProtectedStorageEntry is well formed and valid for the add operation
+        if (!checkPublicKeys(protectedStorageEntry, true) ||
+                !checkSignature(protectedStorageEntry)) {
+
+            log.trace("addProtectedStorageEntry failed due to invalid entry");
+            return false;
+        }
 
         boolean containsKey = map.containsKey(hashOfPayload);
-        if (containsKey) {
-            result = result && checkIfStoredDataPubKeyMatchesNewDataPubKey(protectedStorageEntry.getOwnerPubKey(), hashOfPayload);
+
+        // In a hash collision between two well formed ProtectedStorageEntry, the first item wins and will not be overwritten
+        if (containsKey &&
```

Comment:
> In a hash collision between two well formed ProtectedStorageEntry, the first item wins and will not be overwritten
       
I think that comment is misleading. The checkIfStoredDataPubKeyMatchesNewDataPubKey methods just checks if the pubKey is the same and if the hash is matching. Hash collusion is not an issue as we use cryptographic hashes as keys.

Reply:
Do you have much context on when this condition is expected?

My understanding of the check is to verify that the NewEntry.ownerPubkKey == StoredEntry.ownerPubKey, but how could this fail unless there were identical SeqNr & Payload hashes from different owners? Maybe a misbehaving node sets the value wrong?

I may be missing something really straight forward, but I am happy to update the comment to be more appropriate for the check and not focus on hash collisions.

Reply:
This has been renamed to matchesRelevantPubKey in later patches. The idea is that for ProtectedStorageEntrys we want the Entry.ownerPubKey to match the Payload.ownerPubKey.

For ProtectedMailboxStorageEntrys we want the Entry.receiversPubKey to match the Payload.ownerPubKey.